src/pages/reservationsPage.ts

```diff
@@ -0,0 +1,268 @@
+    if(this.repeatInputValue === ScheduleRepeatOptions.DAILY || this.repeatInputValue === ScheduleRepeatOptions.WEEKLY ){
+      await this.endDateIcon.click();
+      await this.nextMonthButton.click();
+      await this.endDate.click();
+    }
+  }
+
+  async insertStartTime(startTime: string): Promise <void> {
+    if(!this.startCourseCheckboxIsChecked){   
+      await this.startTimeArrowButton.click()
+      await this.page.getByText(`${startTime}`, { exact: true }).click();
+    }
+  }
+
+  async insertEndTime(endTime: string): Promise <void> {
+    if(!this.endCourseCheckboxIsChecked){
+      await this.endTimeArrowButton.click()
+      await this.page.getByText(`${endTime}`, { exact: true }).click();
+    }
+  }
+
+  
+
+  async selectRepeatOption(repeatOption: string): Promise <void> {
+    await this.repeatArrowButton.click();
+    await this.repeatOptionsInput.fill(repeatOption);
+    await this.page.keyboard.press('Enter');
+    this.repeatInputValue = repeatOption
+  }
+
+  async selectDaysToRepeat(dayToRepeat: ScheduleWeekDays): Promise <void> {
+    if(this.repeatInputValue === ScheduleRepeatOptions.WEEKLY){
+      switch (dayToRepeat) {
+        case ScheduleWeekDays.S:
+          await this.page.getByRole('button', { name: `${dayToRepeat}`, exact: true }).click()
+          break;
+        case ScheduleWeekDays.M:
+          await this.page.getByRole('button', { name: `${dayToRepeat}`, exact: true }).click()
+          break;
+        case ScheduleWeekDays.T:
+          await this.page.getByRole('button', { name: `${dayToRepeat}`, exact: true }).click()
+          break;
+        case ScheduleWeekDays.W:
+          await this.page.getByRole('button', { name: `${dayToRepeat}`, exact: true }).click()
+          break;
+        case ScheduleWeekDays.Th:
+          await this.page.getByRole('button', { name: `${dayToRepeat}`, exact: true }).click()
+          break;
+        case ScheduleWeekDays.F:
+          await this.page.getByRole('button', { name: `${dayToRepeat}`, exact: true }).click()
+          break;
+        case ScheduleWeekDays.Sa:
+          await this.page.getByRole('button', { name: `${dayToRepeat}`, exact: true }).click()
+          break;
+      }
+    } 
+  }
+
+  async saveModalChanges(){
+    await this.saveModalChangesButton.click()
+  }
+
+  async clickNewlyCreatedSchedule(schedulName: string): Promise<void>{
+    const elements = await this.allScheduleTitles.all();
+    if(await this.allScheduleTitles.count() !== 0){
+
+      for(const element of elements){
+        if(await element.textContent() === schedulName){
+          await element.click()
+          return;
+        }
+      }
+    }else {
+      throw new Error("schedule not found");
+    }
+  } 
+
+  async openOverideSection(): Promise<void> {
+    this.newOverrideButton.click()
+  }
+
+  async insertOverrideName({overrideName, amount}: rateOverrideData): Promise<void> {
+    await new Promise((resolve) => setTimeout(resolve, 1000));
```

Comment:
do we need to set timeout this way? did you try to use timeout by configuring it in config file?

Reply:
done